### ISSUE TITLE
Improvement: Derive from prog-mode rather than fundamental-mode

### ISSUE THREAD

**Issue by LeonardMH** [jarpy/ks-mode#3]
See [prog-mode on EmacsWiki](https://www.emacswiki.org/emacs/ProgMode). This is a simple change which dramatically improves the functionality of this mode.

- Identifies strings and numbers automatically
- Supports rainbow-delimiters automatically (so `(if (featurep 'rainbow-delimiters) (rainbow-delimiters-mode-enable))` can also be removed from ks.el
- Supports auto-pair for parentheses and braces

**Comment by jarpy** [comment]
Yes. If I remember correctly, I started with fundamental-mode so I could easily reason about what was happening in my code. This should absolutely derive from prog-mode now, though.
